plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/labeling/helper/StyledTextCalculationHelper.xtend

```diff
@@ -198,36 +220,26 @@ class StyledTextCalculationHelper {
 		val definedVariable = variableDeclaration.definedVariable
 		var typeRefString = ""
 
-		 if (definedVariable?.typeRef !== null)
+		if (definedVariable?.typeRef !== null)
 			typeRefString = " : " + getTypeRefDescription(definedVariable.typeRef)
 
-		if (variableDeclaration.declaredTypeRef === null) {
-			styledText = styledText?.append(typeRefString, INFERRED_TYPE_STYLER)
-		} else {
-			styledText = styledText?.append(typeRefString)
-		}
+		styledText = styledText?.append(typeRefString)
 		return styledText;
 	}
 
 	// produces e.g. (param1TypeName, param2TypeName)
-	def private appendStyledTextForFormalParameters(StyledString styledText, FunctionDefinition n4Function,
-		TFunction tFunction) {
+	def private appendStyledTextForFormalParameters(StyledString styledText, TFunction tFunction) {
 		(styledText.append("(") => [
 			if (tFunction.fpars.size > 0) {
-				it.append((0 .. tFunction.fpars.size - 1).map [
-					getStyledTextForFormalParameter(n4Function.fpars.get(it), tFunction.fpars.get(it))
+				it.append((0 .. tFunction.fpars.size - 1).map [i|
```

Comment:
Rewrite this to avoid mapping over index `i`. Map over `fpars` directly instead, i.e. `n4Function.fpars.map`